ui/components/multichain/asset-picker-amount/asset-picker/asset-picker.tsx

```diff
@@ -2,55 +2,69 @@ import React, { useState } from 'react';
 import {
   AvatarTokenSize,
-  Icon,
   IconName,
-  IconSize,
   AvatarToken,
   Text,
   Box,
+  Button,
 } from '../../../component-library';
 import { Asset } from '../../../../ducks/send';
 import {
   AlignItems,
   BackgroundColor,
-  BorderRadius,
   Display,
+  IconColor,
+  JustifyContent,
+  TextColor,
   TextVariant,
 } from '../../../../helpers/constants/design-system';
 import { AssetType } from '../../../../../shared/constants/transaction';
+import { AssetPickerModal } from '../asset-picker-modal/asset-picker-modal';
+import { getNativeCurrency } from '../../../../ducks/metamask/metamask';
 import {
   getIpfsGateway,
   getNativeCurrencyImage,
   getTokenList,
 } from '../../../../selectors';
-import { getNativeCurrency } from '../../../../ducks/metamask/metamask';
-import { AssetPickerModal } from '../asset-picker-modal/asset-picker-modal';
+import Tooltip from '../../../ui/tooltip';
+import { LARGE_SYMBOL_LENGTH } from '../constants';
 import { getAssetImageURL } from '../../../../helpers/utils/util';
 
+export type AssetPickerProps = {
+  asset: Asset;
+  onAssetChange: (newAsset: Asset) => void;
+};
+
 // A component that lets the user pick from a list of assets.
-export default function AssetPicker({ asset }: { asset: Asset }) {
-  const nativeCurrency = useSelector(getNativeCurrency);
-  const nativeCurrencyImage = useSelector(getNativeCurrencyImage);
-  const tokenList = useSelector(getTokenList);
+export function AssetPicker({ asset, onAssetChange }: AssetPickerProps) {
+  const nativeCurrencySymbol = useSelector(getNativeCurrency);
+  const nativeCurrencyImageUrl = useSelector(getNativeCurrencyImage);
+  const tokenList: Record<string, any> = useSelector(getTokenList);
+
   const ipfsGateway = useSelector(getIpfsGateway);
+
   const [showAssetPickerModal, setShowAssetPickerModal] = useState(false);
 
   let image: string | undefined;
+
   if (asset.type === AssetType.native) {
-    image = nativeCurrencyImage;
-  } else if (asset.type === AssetType.token) {
+    image = nativeCurrencyImageUrl;
+  } else if (tokenList && asset.details) {
     image =
-      asset.details?.image ??
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // @ts-ignore: type 'string' can't be used to index type '{}'
-      tokenList?.[asset.details?.address?.toLowerCase()]?.iconUrl;
-  } else if (asset.type === AssetType.NFT) {
-    image = getAssetImageURL(asset.details?.image, ipfsGateway);
+      getAssetImageURL(asset.details?.image, ipfsGateway) ||
+      tokenList[asset.details.address?.toLowerCase()]?.iconUrl;
   }
 
-  // TODO: Handle long symbols in the UI
   const symbol =
-    asset.type === AssetType.native ? nativeCurrency : asset.details?.symbol;
+    asset.type === AssetType.native
+      ? nativeCurrencySymbol
+      : asset.details?.symbol;
+
+  const isSymbolLong = symbol?.length > LARGE_SYMBOL_LENGTH;
+
+  const shortFormSymbol = isSymbolLong
+    ? `${symbol.substring(0, LARGE_SYMBOL_LENGTH - 1)}...`
```

Comment:
Can we set a `max-width` and use `ellipsis` instead?

Reply:
Normally yes, but since we are using a tooltip, this seemed like the best way to ensure the tooltip is consistent with the overflow without using janky `useRef` stuff. Open to thoughts

Reply:
+1, can we use the `title` attr to display untruncated text in a tooltip? I think we should be able to style that as well or assign the same styling as the custom Tooltip